### PR TITLE
Update draft-js to React 16 

### PR DESCRIPTION
**Pull request by johnrjj**
Enables React 16 support when using draft-js.

Upgraded jest, fixed some tests, added some plumbing around testing infra.

Note: Still compatible with React 15.x

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,15 +38,17 @@
     "object-assign": "^4.1.0"
   },
   "peerDependencies": {
-    "react": "^0.14.0 || ^15.0.0-rc",
-    "react-dom": "^0.14.0 || ^15.0.0-rc"
+    "react": "^0.14.0 || ^15.0.0-rc || ^16.0.0-rc || ^16.0.0",
+    "react-dom": "^0.14.0 || ^15.0.0-rc || ^16.0.0-rc || ^16.0.0"
```
Comment (adamsanderson) [inline package.json]: Probably no point in including the release candidate versions, but I guess it doesn't hurt.

**Review by adamsanderson — COMMENTED** [review]
Looks good so far, let's see what shakes out in testing.

File: package.json (hunk 1 above)
Comment (johnrjj) [reply package.json]: Removed the 16rc peer dep

**Comment by johnrjj** [comment]
Ready to merge; /ping @adamsanderson, et al.

This PR is low risk -- only includes testing changes and a tiny fix to className assignment, no logic changes or R16 specific implementation.

**Review by adamsanderson — APPROVED** [review]
👍

**Pull request merged**